Reader: Use new Reader streams discover endpoint

wargcm opened the pull request:
Fixes #22700
WordPressKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/744

## Description

Updates the Discover endpoint to use the new streams API.

## Testing

To test:
- Launch Jetpack and login
- Navigate to the `Discover` feed
- 🔎 **Verify** the first card is a post
- Scroll until you see a blogs recommendation card
- 🔎 **Verify** only three blogs are recommended
- Scroll until you see a tags recommendation card
- 🔎 **Verify** only five tags are recommended

## Regression Notes
1. Potential unintended areas of impact
Should be none. The API is supposed to be a drop-in replacement.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A
PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

wargcm committed:
Add Reader discover endpoint remote feature flag

wargcm committed:
Use new Reader streams discover endpoint

wargcm committed:
Move first card if it's a recommendation card

wargcm committed:
Limit the number of recommended blogs and tags

dvdchr reviewed (approved):
Works as described. LGTM! :shipit:

wargcm committed:
Update mock for Reader discover endpoint

wargcm committed:
Update WordPressKit commit hash

wargcm committed:
Remove redundant tests

These tests were checking parameters passed to the request. In
WordPressKit, there are already tests which check the same thing.

wargcm committed:
Fix tests in `ReaderCardServiceTests` & `ReaderCardTests`

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 500 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dvdchr reviewed (approved):
Apart from the conflict on Podfile.lock, re-approving this again after reviewing the additional changes. Works as described. LGTM :shipit:
